### ISSUE TITLE
shopifyShop name or url?

### ISSUE THREAD

**Issue by silveltman** [thomasKn/astro-shopify#24]
First of all, awesome work!

for the .env file, should shopifyShop be the name or the url? So "my-shop" or "my-shop.myshopify.com"?
I unfortunately get an error for both of them.

Error using name:
```
 error   fetch failed
  File:
    C:\Users\silve\Documents\web-dev\astro\astro-shopify\C:\Users\silve\Documents\web-dev\astro\astro-shopify\node_modules\undici\index.js:109:13
  Stacktrace:
TypeError: fetch failed
    at fetch (C:\Users\silve\Documents\web-dev\astro\astro-shopify\node_modules\undici\index.js:109:13)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async makeShopifyRequest (/src/utils/shopify.ts:38:20)
    at async Module.getProducts (/src/utils/shopify.ts:51:16)
    at async eval (C:\Users\silve\Documents\web-dev\astro\astro-shopify\src\pages\index.astro:20:20)
    at async renderPage (file:///C:/Users/silve/Documents/web-dev/astro/astro-shopify/node_modules/astro/dist/runtime/server/render/page.js:91:30)
    at async renderPage (file:///C:/Users/silve/Documents/web-dev/astro/astro-shopify/node_modules/astro/dist/core/render/core.js:81:20)
    at async renderPage (file:///C:/Users/silve/Documents/web-dev/astro/astro-shopify/node_modules/astro/dist/core/render/dev/index.js:108:10)
    at async handleRoute (file:///C:/Users/silve/Documents/web-dev/astro/astro-shopify/node_modules/astro/dist/vite-plugin-astro-server/route.js:152:20)
    at async run (file:///C:/Users/silve/Documents/web-dev/astro/astro-shopify/node_modules/astro/dist/vite-plugin-astro-server/request.js:46:14)

  Cause:
Error: getaddrinfo ENOTFOUND fulldev-storefront
    at GetAddrInfoReqWrap.onlookup [as oncomplete] (node:dns:71:26)
```

Error using url:
```
 error   Expected object, received null
```

Reproduction:
1. Clone repo
2. Add .env
3. run yarn install
4. run yarn dev

Hope you can help :)

**Comment by thomasKn** [comment]
Hi @silveltman,

Thanks for the feedback!

I just updated the [.env.example](https://github.com/thomasKn/astro-shopify/blob/main/.env.example) file to make it easier for everyone. The `shopifyShop` var should be `your-store.myshopify.com`.

I tried using yarn and everything works on my side.

Do you have products listed in your store? Shopify graphQL API returns null if no ressource is found.

**Comment by silveltman** [comment]
Thanks @thomasKn !

I tried npm and pnpn too, but get the same error. The hard thing is that the error gives very little info.

I do have product in my store. In fact, I just openend a development store with test products. So now I'm thinking about it, here some updated reproduction steps:

1. Go to your Shopify partner dashboard
2. Create a new development store
3. Be sure to select "for testing" on the radio input, instead of "for client"
4. Click "create with test products"
5 previously mentioned steps...

Hope this helps! :)

**Comment by thomasKn** [comment]
@silveltman Did you try adding images to your products?
I'm pretty sure the issue is related to Zod. The [schema utility](https://github.com/thomasKn/astro-shopify/blob/main/src/utils/schemas.ts#L15) doesn't parse Shopify images as optional.
Let me know about the images but I'll make a PR to handle products with no image anyway.

**Comment by silveltman** [comment]
The demo store provides products including images, so I don't think that's the problem here.

**Comment by thomasKn** [comment]
@silveltman do you mind sharing your Storefront API access scopes?
Also, do you use both Public and Private access tokens in your `.env`?

**Comment by thomasKn** [comment]
@silveltman I just created a new store with sample data and everything works on my side.
![image](https://user-images.githubusercontent.com/10447155/226236550-c79f3b63-0446-4757-a008-7b42c222ce90.png)

I also used the [Shopify Headless app](https://apps.shopify.com/headless) to create public and private tokens as follow:
![image](https://user-images.githubusercontent.com/10447155/226236734-ef85f7a7-50fd-4cac-8bfd-838000168b19.png)

By default, the headless app uses all scopes and gives every permissions.

Have you tried the latest version of the main branch?
